### ISSUE TITLE
"caffe.LayerParameter" has no field named "multi_stage_meanfield_param".

### ISSUE THREAD

**Issue by rose0826** [martinkersner/train-CRF-RNN#23]
When I did "python solve.py 2>&1 | tee train.log", this error occurred.
(Corecctly, I followed https://github.com/MasazI/crfasrnn-training and do "./tran.sh".)

WARNING: Logging before InitGoogleLogging() is written to STDERR
I0727 23:16:09.883976  7623 solver.cpp:48] Initializing solver from parameters: 
test_iter: 260
test_interval: 1000
base_lr: 1e-13
display: 50
max_iter: 100000
lr_policy: "fixed"
momentum: 0.99
weight_decay: 0.0005
snapshot: 5000
snapshot_prefix: "models/train_it10"
solver_mode: GPU
net: "TVG_CRFRNN_COCO_VOC_TRAIN_3_CLASSES.prototxt"
I0727 23:16:09.884085  7623 solver.cpp:91] Creating training net from net file: TVG_CRFRNN_COCO_VOC_TRAIN_3_CLASSES.prototxt
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 236:31: Message type "caffe.LayerParameter" has no field named "multi_stage_meanfield_param".
F0727 23:16:09.884837  7623 upgrade_proto.cpp:79] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: TVG_CRFRNN_COCO_VOC_TRAIN_3_CLASSES.prototxt
*** Check failure stack trace: ***
Aborted

Considering caffe version, I changed 
CONVOLUTION to "Convolution", 
POOLING to "Pooling", 
DECONVOLUTION to "Deconvolution", 
CROP to 'Crop',
SPLIT to 'Split', 
DATA to "Data",
ELTWISE to "Eltwise",
DROPOUT to "Dropout",
~ to "SoftmaxWithLoss"
~ to "MultiStageMeanfield"
in all prototxt file of train-CRF-RNN (corecctly, crfasrnn-traing) folder.
However, this error occurred.
What should I do to solve this problem.
Thank you in advance.

**Comment by martinkersner** [comment]
Hi @rose0826,

What does this mean: _~ to "MultiStageMeanfield"_?

Cheers,
Martin

**Comment by rose0826** [comment]
@martinkersner,
Thank you for your reply.
I'm sorry, explanation of my situation  wasn't enough.

That means I changed
type: MULTI_STAGE_MEANFIELD
to
type: "MultiStageMeanfield",
and also, ~ to "SoftmaxWithLoss" means I changed
type: SOFTMAX_LOSS 
to
type: "SoftmaxWithLoss".

And I did what is written in https://github.com/BVLC/caffe/issues/3956
(I forgot to write this and same error occurred, sorry).

Thank you in advance.

**Comment by martinkersner** [comment]
@rose0826,

I guess you have some inconsistency with caffe version, but I don't know what exactly it could be.

Cheers,

Martin

**Comment by rose0826** [comment]
@martinkersner,

Thank you for your reply.
I think so, too.
Maybe, I changed the area that shouldn't be changed in the prototxt.

I think it's better to remove this folder and start all over again.

**Comment by ursular86** [comment]
Hi, I i also have the problem. Have you solved it?

**Comment by chitrang6** [comment]
I have the similar issue. Anyone solved this problem.

**Comment by westeast** [comment]
I have the similar issue. Anyone solved this problem.

**Comment by westeast** [comment]
Net('TVG_CRFRNN_new_deploy.prototxt', 1, weights='TVG_CRFRNN_COCO_VOC.caffemodel')
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 619:31: Message type "caffe.LayerParameter" has no field named "multi_stage_meanfield_param".
F1124 16:02:35.412338 11897 upgrade_proto.cpp:88] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: TVG_CRFRNN_new_deploy.prototxt
*** Check failure stack trace: ***

**Comment by westeast** [comment]
I solve the problem by recompile the https://github.com/bittnt/caffe caffe

**Comment by DinhOsh** [comment]
I have also got this error.
[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 619:31: Message type "caffe.LayerParameter" has no field named "multi_stage_meanfield_param".
F0524 16:07:38.742789 39956 upgrade_proto.cpp:88] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: TVG_CRFRNN_new_deploy.prototxt
*** Check failure stack trace: ***

I am looking for someone to help me to resolve this problem.
Thanks

**Comment by KleinYuan** [comment]
@DinhOsh 
change `multi_stage_meanfield_param`  to `MultiStageMeanfield`

**Comment by Sam813** [comment]
@DinhOsh I have the same problem. Have you found any solution for it?
